src/Inkluzitron/Modules/Urban/UrbanModule.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using Inkluzitron.Contracts;
+using Inkluzitron.Extensions;
+using Inkluzitron.Models;
+using Inkluzitron.Models.Settings;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Inkluzitron.Modules.Urban
+{
+    [Name("Urban Dictionary")]
+    public class UrbanModule : ModuleBase, IReactionHandler
+    {
+        private static List<UrbanQueryResult> UrbanResultsCache = new ();
+
+        private string ApiUrl { get; }
+        private string UrbanEmbedLogo { get; }
+        private string UrbanNotFound { get; }
+        private IHttpClientFactory HttpClientFactory { get; }
+        private ReactionSettings ReactionSettings { get; }
+        private DiscordSocketClient Client { get; }
+
+        public UrbanModule(IConfiguration config, DiscordSocketClient client,
+            IHttpClientFactory httpClientFactory, ReactionSettings reactionSettings)
+        {
+            Client = client;
+            ApiUrl = config["UrbanApiUrl"];
+            UrbanEmbedLogo = config["UrbanEmbedLogo"];
+            UrbanNotFound = config["UrbanNotFound"];
+            HttpClientFactory = httpClientFactory;
+            ReactionSettings = reactionSettings;
+        }
+
+        private async Task<UrbanQueryResult> GetDefinitions(string query)
+        {
+            var cachedResult = UrbanResultsCache.FirstOrDefault(r => r.Query == query);
+            if (cachedResult != null) return cachedResult;
+
+            var response = await HttpClientFactory.CreateClient()
+                .GetAsync(string.Format(ApiUrl, HttpUtility.UrlEncode(query)));
+
+            var responseData = await response.Content.ReadAsStringAsync();
+            var result = JsonConvert.DeserializeObject<UrbanQueryResult>(responseData);
+
+            result.Query = query;
+
+            UrbanResultsCache.Add(result);
+            return result;
+        }
+
+        [Command("urban")]
+        [Alias("define", "wtf")]
```

Comment:
To `wtf` je takové wtf. Pokud to bude někdo interpretovat jako na fitwide, tak tam je to pro zjištění předmětu. Může to být matoucí.